src/details/ArborX_DetailsDendrogram.hpp

```diff
@@ -0,0 +1,158 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_DENDROGRAM_HPP
+#define ARBORX_DETAILS_DENDROGRAM_HPP
+
+#include <ArborX_DetailsKokkosExtSwap.hpp>
+#include <ArborX_DetailsKokkosExtViewHelpers.hpp>
+#include <ArborX_DetailsSortUtils.hpp>
+#include <ArborX_DetailsUnionFind.hpp>
+#include <ArborX_MinimumSpanningTree.hpp> // WeightedEdge
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Details
+{
+
+// Compute permutation that orders edges in increasing order
```

Comment:
```suggestion
// Compute permutation that sorts edges in increasing weight order
```
We should probably comment that permutation is non-deterministic and why that is fine

Reply:
By non-deterministic, do you mean that we it's not stable?

Reply:
Yes, if weights are not unique